### PR TITLE
Add org.kde.lokalize (Lokalize)

### PR DESCRIPTION
**Pull request by cho2**
Add org.kde.lokalize (Lokalize)

### CODE REVIEW

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
Cuplikan layar dari 2020-06-20 22-14-03

LGTM

File: org.kde.lokalize.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                }
+            ]
+        },
+        {
+            "name": "hunspell",
+            "sources": 
+            [
+                { 
+                    "type": "git",
+                    "url": "https://github.com/hunspell/hunspell.git"
```
Comment (nedrichards) [inline org.kde.lokalize.json:41]: what version here? you need to pin to a specific version, not just use master.

**Comment by tsdgeos** [comment]
This is going to need host filesystem access, otherwise the project functionality is not going to work.

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
bot, build org.kde.lokalize

File: org.kde.lokalize.json (hunk 1 above)
Comment (cho2) [reply org.kde.lokalize.json]: https://github.com/flathub/flathub/pull/1603/commits/5cd008c9be83162308f1a0b7d32c91eb853ae475

**Comment by cho2** [comment]
recheck

Cuplikan layar dari 2020-06-21 14-33-13

LGTM

File: org.kde.lokalize.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            "name": "kross",
+            "buildsystem": "cmake-ninja",
+            "config-opts": 
+            [ 
+                "-DENABLE_TESTING=OFF"
+            ],
+            "sources": 
+            [ 
+                { 
+                    "type": "git",
```
Comment (tsdgeos) [inline org.kde.lokalize.json:30]: any particular reason for using the git repo here instead of the tarball?

File: appdata.patch [hunk 3]
```diff
@@ -0,0 +1,8 @@
+--- a/org.kde.lokalize.appdata.xml	2020-06-09 01:03:51.000000000 +0700
++++ b/org.kde.lokalize.appdata.xml	2020-06-20 21:45:11.981635367 +0700
+@@ -139,4 +139,5 @@
+     <release version="20.04.0" date="2020-04-23"/>
+     <release version="19.12.1" date="2020-01-09"/>
+   </releases>
++  <content_rating type="oars-1.1" />
```
Comment (tsdgeos) [inline appdata.patch:7]: Could you submit a Merge Request upstream about this?

https://invent.kde.org/sdk/lokalize

File: org.kde.lokalize.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
+{
+    "id": "org.kde.lokalize",
+    "branch": "master",
```
Comment (bilelmoussaoui) [inline org.kde.lokalize.json:3]:
```suggestion
```

File: org.kde.lokalize.json (hunk 2 above)
Comment (cho2) [reply org.kde.lokalize.json:30]: I did not find it on https://invent.kde.org/frameworks/kross/-/releases. Can I use from https://github.com/KDE/kross/releases ?

File: appdata.patch (hunk 3 above)
Comment (cho2) [reply appdata.patch:7]: https://invent.kde.org/sdk/lokalize/-/merge_requests/1

File: org.kde.lokalize.json (hunk 4 above)
Comment (cho2) [reply org.kde.lokalize.json]: https://github.com/flathub/flathub/pull/1603/commits/f653cfbf4e56f58657298fbf9afbf9703e27fc3d

File: org.kde.lokalize.json (hunk 2 above)
Comment (tsdgeos) [reply org.kde.lokalize.json:29]: Don't use github, it makes no sense, it's just a mirror. Use https://download.kde.org/stable/frameworks/

File: appdata.patch (hunk 3 above)
Comment (tsdgeos) [reply appdata.patch:7]: Ideally you would have made that against release/20.04 so it gets included in 20.04.3 released next month, but that's fine, i guess we can wait carrying the patch until 20.08

File: org.kde.lokalize.json (hunk 2 above)
Comment (cho2) [reply org.kde.lokalize.json:29]: Can not found Kross under https://download.kde.org/stable/frameworks/

Comment (tsdgeos) [reply org.kde.lokalize.json:29]: Sorry forgot to mention it's inside portingAids, i.e.

https://download.kde.org/stable/frameworks/5.69/portingAids/kross-5.69.0.tar.xz

**Comment by cho2** [comment]
bot, build org.kde.lokalize

File: org.kde.lokalize.json (hunk 2 above)
Comment (cho2) [reply org.kde.lokalize.json]: https://github.com/flathub/flathub/pull/1603/commits/9ca1a3efba615ca5e3bf82323c3b5f8d2de30e17

File: org.kde.lokalize.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            [ 
+                { 
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/frameworks/5.69/portingAids/kross-5.69.0.tar.xz",
+                    "sha256": "acec70fe97dddd32b05b28747da7761571d8b684641c7848f544a0e67a698f93"
+                }
+            ]
+        },
+        {
+            "name": "hunspell",
```
Comment (tsdgeos) [inline org.kde.lokalize.json:36]: Do we need this?

If i do flatpak run org.kde.Sdk i can see /usr/lib/x86_64-linux-gnu/libhunspell-1.7.so is there so it seems hunspell is part of the SDK and we should not need to compile it?

Comment (cho2) [reply org.kde.lokalize.json:36]: I'll check it

**Comment by cho2** [comment]
bot, build org.kde.lokalize

File: org.kde.lokalize.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    "rename-icon": "lokalize",
+    "finish-args": 
+    [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--filesystem=host"
+    ],
+    "separate-locales": false,
```
Comment (tsdgeos) [inline org.kde.lokalize.json:16]: I don't think we use this in any of the org.kde.XXX flathubs, any special reason for this one to be different?

Comment (cho2) [reply org.kde.lokalize.json:16]: Still based on https://invent.kde.org/kukuhsyafaat/flatpak-kde-applications/-/blob/master/org.kde.lokalize.json#L11, try to remove and check it

**Comment by cho2** [comment]
bot, build org.kde.lokalize

File: org.kde.lokalize.json (hunk 5 above)
Comment (cho2) [reply org.kde.lokalize.json]: https://github.com/flathub/flathub/pull/1603/commits/5057c7ae60aa3ad8892d5d486ee867be72622819

File: org.kde.lokalize.json (hunk 6 above)
Comment (cho2) [reply org.kde.lokalize.json]: https://github.com/flathub/flathub/pull/1603/commits/0c2ab283f731a6ba5badc0935d188357c8f0630a

**Comment by cho2** [comment]
Cuplikan layar dari 2020-06-26 13-11-33

Lokalize run, but can't open po file

**Comment by tsdgeos** [comment]
Works fine for me

imatge

**Comment by tsdgeos** [comment]
@cho2 I'm confused becasue you had it working before, then it broke? Can you figure out why/when it broke for you?

**Comment by cho2** [comment]
I'll check the last build later

**Comment by cho2** [comment]
Cuplikan layar dari 2020-06-29 08-57-26

LGTM

**Comment by tsdgeos** [comment]
@nedrichards @bilelmoussaoui Anything else you can see is missing or can it be created?

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.lokalize

**Comment by bilelmoussaoui** [comment]
As the application allows the user to select a default language. We need the `seperate-locale: false` back in the manifest. Otherwise looks good to me 

image

**Comment by tsdgeos** [comment]
Why would seperate-locale make a difference?

Doesn't that just say if the locale is in a different package or not?

I still get both installed when they are separate

```
$ flatpak install --user https://dl.flathub.org/build-repo/21708/org.kde.lokalize.flatpakref

org.kde.lokalize permissions:
    ipc   wayland   x11   dri   file access [1]   dbus access [2]

    [1] host, xdg-config/kdeglobals:ro
    [2] com.canonical.AppMenu.Registrar


        ID                                Branch           Op           Remote                    Download
 1. [✓] org.kde.lokalize.Locale           test             i            lokalize-origin           69,3 kB / 1,3 MB
 2. [✓] org.kde.lokalize                  test             i            lokalize-origin            2,1 MB / 2,4 MB

```

**Comment by bilelmoussaoui** [comment]
The `seperate-locale` disables having just the language used by the user being downloaded/installed in the `.Locale` extension. You still get a `.Lcoale` extension, you will just have to download all the languages

**Comment by cho2** [comment]
Should bring `seperate-locale: false` back?

**Comment by tsdgeos** [comment]
Yes please, i'm not 100% sure it'll fix the issue @bilelmoussaoui found about not getting the list of languages but let's try :)

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by nedrichards** [comment]
So I could switch app language using Help -> Switch Application Language. The "default language" setting mentioned above didn't populate out of the box but I'm not sure if further setup is required here.

> org.kde.kross.core: Action::setInterpreter: interpreter not found:  "python"

The runtime provides `python3` rather than `python`.

> AL lib: (EE) ALCplaybackOSS_open: Could not open /dev/dsp: No such file or directory

That's fixed by adding the `--socket=pulseaudio` permission.

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by tsdgeos** [comment]
> The "default language" setting mentioned above didn't populate out of the box but I'm not sure if further setup is required here.

As i expected :)

@cho2 Please add kdelibs4support to the build, that will fix that combobox.

You can find it in https://download.kde.org/stable/frameworks/5.69/portingAids/

**Comment by tsdgeos** [comment]
FWIW added an upstream bug at https://bugs.kde.org/show_bug.cgi?id=423772

**Comment by cho2** [comment]
bot, build org.kde.lokalize

**Comment by cho2** [comment]
Cuplikan layar dari 2020-07-02 09-46-22

LGTM

**Comment by nedrichards** [comment]
Cool, is the python vs python3 thing fine? A cursory search through the gitlab suggested that there were a few cases of just straight up calling python, as well as some where there was a fallthrough to python3 if it wasn't present - but I didn't do a big investigation. Other than that, lgtm.

**Comment by tsdgeos** [comment]
> Cool, is the python vs python3 thing fine? A cursory search through the gitlab suggested that there were a few cases of just straight up calling python, as well as some where there was a fallthrough to python3 if it wasn't present - but I didn't do a big investigation. Other than that, lgtm.

That's "not fixable". kross only works with python2, so just ignore for now, i hope eventually we'll port kross to python3

**Comment by tsdgeos** [comment]
I'd prefer if we removed the separate-locales line like the other json files we use for org.kde.XXX but we've danced around that for like 10 times already so i'm not going to pretend i care that much.

Can we get it merged?

**Comment by bilelmoussaoui** [comment]
> > Cool, is the python vs python3 thing fine? A cursory search through the gitlab suggested that there were a few cases of just straight up calling python, as well as some where there was a fallthrough to python3 if it wasn't present - but I didn't do a big investigation. Other than that, lgtm.
> 
> That's "not fixable". kross only works with python2, so just ignore for now, i hope eventually we'll port kross to python3

We can bundle python2 in the mean time? it's available in shared-modules.



> I'd prefer if we removed the separate-locales line like the other json files we use for org.kde.XXX but we've danced around that for like 10 times already so i'm not going to pretend i care that much.

Sure we can remove that, just expect issues that the language selector doesn't work.

**Comment by tsdgeos** [comment]
> > > Cool, is the python vs python3 thing fine? A cursory search through the gitlab suggested that there were a few cases of just straight up calling python, as well as some where there was a fallthrough to python3 if it wasn't present - but I didn't do a big investigation. Other than that, lgtm.
> > 
> > 
> > That's "not fixable". kross only works with python2, so just ignore for now, i hope eventually we'll port kross to python3
> 
> We can bundle python2 in the mean time? it's available in shared-modules.

I guess that's possible, it's not as if the functionality lost is huge, but won't hurt.

> 
> > I'd prefer if we removed the separate-locales line like the other json files we use for org.kde.XXX but we've danced around that for like 10 times already so i'm not going to pretend i care that much.
> 
> Sure we can remove that, just expect issues that the language selector doesn't work.

Which language selector? The one in the config dialog has nothing to do with that and is fixed by the dependency on kdelibs4support

**Comment by bilelmoussaoui** [comment]
> > Sure we can remove that, just expect issues that the language selector doesn't work.
> 
> Which language selector? The one in the config dialog has nothing to do with that and is fixed by the dependency on kdelibs4support

The selector working doesn't mean the locale files will be available.

**Comment by tsdgeos** [comment]
Do you even understand what that selector is for? It's not to change the application language, it's to set the default language the translator is using to translate .po files.

You don't need any locale file for that to work, do you?

**Comment by tsdgeos** [comment]
Re-reading my last comment, it came out more aggressive than it should have been. Please accept my apologies.

**Comment by nedrichards** [comment]
> > > > Cool, is the python vs python3 thing fine? A cursory search through the gitlab suggested that there were a few cases of just straight up calling python, as well as some where there was a fallthrough to python3 if it wasn't present - but I didn't do a big investigation. Other than that, lgtm.
> > > 
> > > 
> > > That's "not fixable". kross only works with python2, so just ignore for now, i hope eventually we'll port kross to python3
> > 
> > 
> > We can bundle python2 in the mean time? it's available in shared-modules.
> 
> I guess that's possible, it's not as if the functionality lost is huge, but won't hurt.

Not a blocker for now then, will merge.

**Pull request closed without merging**